meta-mender-core/recipes-update-modules/mender-app-update-module/mender-app-update-module_git.bb

```diff
@@ -0,0 +1,53 @@
+DESCRIPTION = "Mender Application Update Module for supporting containerized application updates on devices including related tools."
+
+HOMEPAGE = "https://mender.io"
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=b4b4cfdaea6d61aa5793b92efd42e081"
+
+SRC_URI = "git://github.com/mendersoftware/app-update-module;branch=master;protocol=https"
+
+SRCREV = "15025d9f1e21eafcbf1325b2a5034b2606e1d719"
+
+S = "${WORKDIR}/git"
+
+RDEPENDS:${PN} = "jq mender-client xdelta3"
+
+PACKAGECONFIG ?= ""
+PACKAGECONFIG[docker-compose] = ",,,docker-compose"
+
+inherit allarch
+
+do_configure[noexec] = "1"
+do_compile[noexec] = "1"
+
+do_install:class-target() {
+    # install the Application Update Module
+    install -d ${D}/${datadir}/mender/app-modules/v1
+    install -d ${D}/${datadir}/mender/modules/v3
+    install -m 755 ${S}/src/app ${D}/${datadir}/mender/modules/v3/app
+
+    # install the Docker Compose Module
+    install -m 755 ${S}/src/app-modules/docker-compose ${D}/${datadir}/mender/app-modules/v1/docker-compose
```

Comment:
Are you sure? Shouldn't we install all the subsystems all the times? @merlin-northern 
From my PoV, the update module includes everything and the PACKAGECONFIG controls dependencies.
These are not build options.

Reply:
In the Yocto mindset I am sure. `PACKAGECONFIG` is the management for build options.

Reply:
I am having doubts about requiring the underlying orchestrator during the build. I can run my custom containers setup, which is build from custom recipes which provide all I want and all I need to run application updates based on containerized application.

Reply:
> I am having doubts about requiring the underlying orchestrator during the build.

I believe they are specified as a runtime dependency and that take place in the PACKAGECONFIG definition.

```
                           #Flag for enabled    #Flag for disabled  #Build dependency   #Runtime dependency
PACKAGECONFIG[docker-compose] = "[1], 			[2],             [3],               docker-compose"
```

So as long as the custom recipe provides an equivalent orcastrator as the default `docker-compose`, `k3s`, k8s` (so the recipe specifies a custom RPROVIDES). This will still work.

The argument for the conditional install  approach is that there is more value to have simpler steps to make it work for the people with the default case (they won't have a successful build only to find out things don't work at runtime) versus a super flexible setup for an edge case - which can still be achieved but need the correct RDEPENDS in the custom recipes.